app/models/organization.rb

```diff
@@ -16,23 +16,33 @@ class Organization < ApplicationRecord
 
   # ----------------
   # Region callbacks
+  #
+  # * These callbacks are medium-term temporary.
+  # * This class and the Region callbacks should ideally be totally superseded by the Region class.
+  # * Keep the callbacks simple (avoid branching and optimization), idempotent (if possible) and loud when things break.
+  #
+  # - kit (11/2020)
   after_create :create_region, if: -> { Flipper.enabled?(:regions_prep) }
-  before_update :update_region, if: -> { Flipper.enabled?(:regions_prep) }
+  after_update :update_region, if: -> { Flipper.enabled?(:regions_prep) }
 
   def create_region
     return if region&.persisted?
-    parent = Region.find_by!(region_type: Region.region_types[:root])
-    region = build_region(name: name, description: description, reparent_to: parent)
-    region.region_type = Region.region_types[:organization]
-    region.save!
+
+    create_region!(
+      name: name,
+      description: description,
+      reparent_to: Region.root,
+      region_type: Region.region_types[:organization]
+    )
   end
 
   def update_region
-    return unless name_changed? || description_changed?
```

Comment:
This is an unnecessary optimization that makes this callback more complicated than it needs to.

Reply:
I'm curious - was it causing any problems?

Reply:
Not particularly, it just seemed like a _feel-good_ check rather than adding any serious value. Besides, since it's now an `after_update` callback, I don't think dirty is expected to work. And it seems like a better fit for an `after_*` callback semantically.

Also, I think if we were to defend against unnecessary updates, it should ideally be written as:

```ruby
if name_changed?
  region.name = name
elsif description_changed?
  region.description = description
end

region.save!
```

Which just seems too much for a really simple idempotent operation. 

It also tends to make it not clear whether it's happening because we want to avoid unnecessary updates or if there's a hard-constraint because of which this shouldn't be run. 

In my mind, it's simpler for me to think about this as: _no matter what, region gets updated_.